testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OfflineTokenTest.java

```diff
@@ -255,24 +256,24 @@ public void offlineTokenBrowserFlow() throws Exception {
         setTimeOffset(3000000);
 
         OAuthClient.AccessTokenResponse response = oauth.doRefreshTokenRequest(newRefreshTokenString, "secret1");
+        RefreshToken newRefreshToken = oauth.parseRefreshToken(newRefreshTokenString);
         Assert.assertEquals(400, response.getStatusCode());
         assertEquals("invalid_grant", response.getError());
 
-        events.expectRefresh(offlineToken.getId(), sessionId)
+        events.expectRefresh(offlineToken.getId(), newRefreshToken.getSessionState())
                 .client("offline-client")
                 .error(Errors.INVALID_TOKEN)
                 .user(userId)
                 .clearDetails()
                 .assertEvent();
 
-
         setTimeOffset(0);
     }
 
     private String testRefreshWithOfflineToken(AccessToken oldToken, RefreshToken offlineToken, String offlineTokenString,
                                                final String sessionId, String userId) {
         // Change offset to big value to ensure userSession expired
-        setTimeOffset(99999);
+        setTimeOffset(30000);
```

Comment:
I am not 100% sure why this change is needed? AFAIK the point of this test is, that "refresh of offline token" is possible even if the "online" user session (which was involved in the original creation of offline session) already expired and is removed in Keycloak. Lowering offset to 30000 means that "online" session may still remain in the cache (as max timeout for online session is 10 hours), which should not be needed and that is exactly the point of this test (Make sure that refresh with offline token works even if all storages of corresponding "online" session are completely removed in Keycloak).

Reply:
You are right. This change is not needed so I reverted it where ever I did this kind of change.